### PR TITLE
feat(config): add config service to manage the application

### PR DESCRIPTION
**Pull request by mbarbeau**

### CODE REVIEW

File: src/demo-app/app/module.ts [hunk 1]
```diff
@@ -34,6 +36,10 @@ export function languageLoader(http: Http) {
     IgoModule.forRoot()
   ],
   providers: [
+    provideConfigOptions({
+      default: environment.igo,
+      jsonPathFile: './config/config.json'
```
Comment (cbourget) [inline src/demo-app/app/module.ts]: Should be "jsonFilePath" or simply "path"

File: src/lib/core/config/config.interface.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface ConfigOptions {
+  default?: {[key: string]: any},
+  jsonPathFile?: string
```
Comment (cbourget) [inline src/lib/core/config/config.interface.ts]: See comment above

File: src/lib/core/config/config.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+
+import { ObjectUtils } from "../../utils";
```
Comment (cbourget) [inline src/lib/core/config/config.service.ts]: Single quotes

File: src/lib/search/search-sources/icherche-search-source.ts [hunk 4]
```diff
@@ -14,13 +14,15 @@ import { SearchSourceOptions } from './search-source.interface';
 export class IChercheSearchSource extends SearchSource {
 
   static _name: string = 'ICherche Québec';
-  static searchUrl: string = 'https://geoegl.msp.gouv.qc.ca/icherche/geopasdecode';
+  private searchUrl: string = 'https://geoegl.msp.gouv.qc.ca/icherche/geopasdecode';
 
   constructor(private jsonp: Jsonp,
               @Inject(SEARCH_SOURCE_OPTIONS)
-              private options: SearchSourceOptions) {
+              private options: SearchSourceOptions,
```
Comment (cbourget) [inline src/lib/search/search-sources/icherche-search-source.ts]: Should search options such as "limit" be configurable in the service?

Comment (mbarbeau) [inline src/lib/search/search-sources/icherche-search-source.ts]: Yes and several other options in other services.

**Pull request merged**